processing/src/main/java/org/apache/druid/query/QueryRunnerFactoryConglomerate.java

```diff
@@ -21,7 +21,9 @@
 
 /**
  */
-public interface QueryRunnerFactoryConglomerate
+public interface QueryRunnerFactoryConglomerate extends QueryToolChestWarehouse
```

Comment:
I believe 2 things needs to be done here:
* remove  `extends QueryToolChestWarehouse` and use `QueryRunnerFactoryConglomerate`
* possibly remove the `QueryLogic` to be tied to the `QueryToolChest` like it is now - so it will be valid to only specify that
   *  possibly introduce a new "map" for these things?

these will cause some refactors to happen - mostly in tests; I would like to postpone that to a followup PR